cpp/open3d/ml/contrib/RoIPoolKernel.cu

```diff
@@ -0,0 +1,300 @@
+
+    if (sample_pt_idx >= sampled_pts_num || box_idx >= boxes_num ||
+        bs_idx >= batch_size) {
+        return;
+    }
+
+    if (pooled_empty_flag[bs_idx * boxes_num + box_idx]) {
+        return;
+    }
+
+    int temp_idx = bs_idx * boxes_num * sampled_pts_num +
+                   box_idx * sampled_pts_num + sample_pt_idx;
+    int src_pt_idx = pts_idx[temp_idx];
+    int dst_feature_offset = temp_idx * (3 + feature_in_len);
+
+    for (int j = 0; j < 3; j++)
+        pooled_features[dst_feature_offset + j] =
+                xyz[bs_idx * pts_num * 3 + src_pt_idx * 3 + j];
+
+    int src_feature_offset =
+            bs_idx * pts_num * feature_in_len + src_pt_idx * feature_in_len;
+    for (int j = 0; j < feature_in_len; j++)
+        pooled_features[dst_feature_offset + 3 + j] =
+                pts_feature[src_feature_offset + j];
+}
+
+void roipool3dLauncher(int batch_size,
+                       int pts_num,
+                       int boxes_num,
+                       int feature_in_len,
+                       int sampled_pts_num,
+                       const float *xyz,
+                       const float *boxes3d,
+                       const float *pts_feature,
+                       float *pooled_features,
+                       int *pooled_empty_flag) {
+    // printf("batch_size=%d, pts_num=%d, boxes_num=%d\n", batch_size, pts_num,
+    // boxes_num);
+    int *pts_assign = NULL;
+    cudaMalloc(&pts_assign, batch_size * pts_num * boxes_num *
+                                    sizeof(int));  // (batch_size, N, M)
+    // cudaMemset(&pts_assign, -1, batch_size * pts_num * boxes_num *
+    // sizeof(int));
+
+    dim3 blocks(DIVUP(pts_num, THREADS_PER_BLOCK), boxes_num,
+                batch_size);  // blockIdx.x(col), blockIdx.y(row)
+    dim3 threads(THREADS_PER_BLOCK);
+    assign_pts_to_box3d<<<blocks, threads>>>(batch_size, pts_num, boxes_num,
+                                             xyz, boxes3d, pts_assign);
+
+    int *pts_idx = NULL;
+    cudaMalloc(&pts_idx,
+               batch_size * boxes_num * sampled_pts_num *
+                       sizeof(int));  // (batch_size, M, sampled_pts_num)
+
+    dim3 blocks2(DIVUP(boxes_num, THREADS_PER_BLOCK),
+                 batch_size);  // blockIdx.x(col), blockIdx.y(row)
+    get_pooled_idx<<<blocks2, threads>>>(batch_size, pts_num, boxes_num,
+                                         sampled_pts_num, pts_assign, pts_idx,
+                                         pooled_empty_flag);
+
+    dim3 blocks_pool(DIVUP(sampled_pts_num, THREADS_PER_BLOCK), boxes_num,
+                     batch_size);
+    roipool3d_forward<<<blocks_pool, threads>>>(
+            batch_size, pts_num, boxes_num, feature_in_len, sampled_pts_num,
+            xyz, pts_idx, pts_feature, pooled_features, pooled_empty_flag);
+
+    cudaFree(pts_assign);
+    cudaFree(pts_idx);
+
+#ifdef DEBUG
+    cudaDeviceSynchronize();  // for using printf in kernel function
+#endif
+}
```

Comment:
Add a newline at the end. Without the newline, github shows as a "stop" sign.

![image](https://user-images.githubusercontent.com/1501945/107713214-94524c80-6c7f-11eb-8d49-c2b521cf8481.png)

Same for other files that don't have the newline at the end.